mantidproject/mantid
# Inelastic Settings Developer feature flags option

### Description of work
This PR adds a new option to the Inelastic settings widget called "Developer feature flags". This will make it easier to get feedback from advanced software users through the use of rapid prototyping and receiving early feedback without allowing an unfinished feature to be fully accessible to all users of the software immediately. It will shorten the feedback loop and should hopefully lead to software which more closely aligns with user wishes.

Fixes #36533

### To test:
1. Open Inelastic Data Analysis interface
2. Click the settings cog in the bottom left
3. Add a string to the `Developer feature flags` option. e.g. 'sqw' . You can add multiple flags by separating them by a space
4. Apply and close the settings
5. Open the settings again, you can see the flag you just set should be saved
6. Change the `Developer feature flags` again. Close the settings and the interface itself, and the workbench
7. Reopen the workbench and open Inelastic Data Analysis. Open the settings. You should see the option should still be saved to its previous value

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
